### ISSUE TITLE
Fully connected working better than CNN?

### ISSUE THREAD

**Issue by Cassiano-Silva** [aamini/introtodeeplearning#50]
Hi there,

it's more a question, not sure if it's an issue.

In Lab2 Part 1, two network types are analyzed: Fully Connected and CNN.
testing both with the test images show much better results with Fully then with CNN.
I tried changing parameters (learning rate and optimizer) but it didn't change so much the results.

CNN showed 8 correct estimations out of 20 test images.
Fully showed 19 correct estimations out of 20 test images.

I was expecting CNN to show better results, I thought it was more appropriate for vision applications. 
Did I do something wrong?

Thanks and regards,

Cassiano

**Comment by NiranthS** [comment]
This dataset is a very small one 28X28(ie 784 pixels) which is very less as compared to the pictures we use daily.
If you use the same dense network for a 256X256 image, you will see the performance of CNN is far better than a simple fully connected network.
Even in MNIST, with proper optimizer and loss, CNN works slightly better than fully connected.
(Use optimizer='adam', loss='sparse_categorical_crossentropy')

**Comment by aamini** [comment]
Echoing @NiranthS's comments. For larger images the difference will be even larger but even for MNIST you should be seeing a fairly large difference already. Your results 8/20 for CNN and 19/20 for Dense indicate that you have an bug in your code for the CNN model or training. I suggest debugging the CNN to see where the error is.
